src/components/fragment.js

```diff
@@ -17,111 +73,78 @@ type Props = {
   children: React.Element<*>
 };
 
-const relativePaths = (ComposedComponent: ReactClass<*>) => {
-  class RelativeFragment extends Component {
-    constructor() {
-      super();
-      this.id = generateId();
-    }
+class Fragment extends Component {
+  matcher: ?Object;
 
-    getChildContext() {
-      const { parentRoute } = this.context;
-      const { forRoute } = this.props;
-
-      return {
-        // Append the parent route if this isn't the first
-        // RelativeFragment in the hierarchy.
-        parentRoute: parentRoute &&
-          parentRoute !== '/' &&
-          parentRoute !== forRoute
-          ? `${parentRoute}${forRoute || ''}`
-          : forRoute,
-        parentId: this.id
-      };
-    }
+  constructor(props: Props, context: typeof Fragment.contextTypes) {
+    super(props, context);
```

Comment:
You also don't need to pass `props`, unless you're accessing `this.props` in the constructor. React will populate them both on the instance for you.